datafusion/optimizer/src/eliminate_cross_join.rs

```diff
@@ -96,6 +96,7 @@ impl OptimizerRule for EliminateCrossJoin {
                 filter.input.as_ref(),
                 LogicalPlan::Join(Join {
                     join_type: JoinType::Inner,
+                    null_equals_null: false,
```

Comment:
And also, not sure that this rule shouldn't have any interactions with `null_equals_null = true` joins -- what is the current behavior of such queries and won't this additional filter change it drastically?